Proxies/Numbers.cs

```diff
@@ -29,7 +29,7 @@ public abstract class IntegerProxy {
         }
 
         [JSIsPure]
-        [JSReplacement("($this === $rhs)")]
+        [JSReplacement("(($this).valueOf() === ($rhs).valueOf())")]
```

Comment:
I think .valueOf() will force the creation of an object wrapper, so you probably want to do ```x | 0 === y | 0``` for signed ints, ```x >>> 0 === y >>> 0``` for unsigned ints, and ```+x === +y``` for floats.